platform/mbed_error.h

```diff
@@ -60,7 +67,17 @@
 extern "C" {
 #endif
 
-void error(const char* format, ...);
+
+static inline void error(const char *format, ...) {
+#ifndef NDEBUG
+    va_list arg;
+    va_start(arg, format);
+    mbed_error_vfprintf(format, arg);
+    va_end(arg);
+#endif
+    while (1) {}
```

Comment:
`exit` flush the file standard IOs. Because `exit` reference them, a lot of code from the IO subsystem is pulled in even if IOs function are not used directly in the actual program. 

I've made a PR 6 month ago which address this issue: #2741 . 
It allows user to disable flush of standard IOs in mbed_app.json.